ui/app/AppLayouts/Chat/ContactsColumn/ChannelList.qml

```diff
@@ -83,17 +83,66 @@ ScrollView {
-
-            // TODO implement mute chat in Model and call it here
-            Action { text: qsTr("15 minutes"); icon.width: 0; }
-            Action { text: qsTr("1 hour"); icon.width: 0; }
-            Action { text: qsTr("8 hours"); icon.width: 0; }
-            Action { text: qsTr("24 hours"); icon.width: 0; }
-            Action { text: qsTr("Until I turn it back on"); icon.width: 0; }
+        Action {
+            text: chatsModel.channelIsMuted(channelContextMenu.channelIndex) ? qsTr("Unmute chat") : qsTr("Mute chat")
+            icon.source: "../../../img/bell.svg"
+            icon.width: 16
+            icon.height: 16
+            onTriggered: {
+              if (chatsModel.channelIsMuted(channelContextMenu.channelIndex)) {
+                chatsModel.unmuteChannel(channelContextMenu.channelIndex)
+                return
+              }
+              chatsModel.muteChannel(channelContextMenu.channelIndex)
+            }
         }
+
+        /* PopupMenu { */
+        /*     hasArrow: false */
+        /*     title: qstr("Mute chat") */
+
+        /*     // TODO implement mute chat in Model and call it here */
+        /*     Action { */ 
+        /*         text: qsTr("15 minutes"); */
+        /*         icon.width: 0; */ 
+        /*         onTriggered: { */
+        /*             chatsModel.muteChannel(channelContextMenu.channelIndex, Constants.muteChat15Minutes) */
+        /*         } */
+        /*     } */
+        /*     Action { */
+        /*         text: qsTr("1 hour"); */
+        /*         icon.width: 0; */
+        /*         onTriggered: { */
+        /*             chatsModel.muteChannel(channelContextMenu.channelIndex, Constants.muteChatOneHour) */
+        /*         } */
+        /*     } */
+        /*     Action { */
+        /*         text: qsTr("8 hours"); */
+        /*         icon.width: 0; */
+        /*         onTriggered: { */
+        /*             chatsModel.muteChannel(channelContextMenu.channelIndex, Constants.muteChatEightHours) */
+        /*         } */
+        /*     } */
+        /*     Action { */ 
+        /*         text: qsTr("24 hours"); */ 
+        /*         icon.width: 0; */
+        /*         onTriggered: { */
+        /*             chatsModel.muteChannel(channelContextMenu.channelIndex, Constants.muteChat24Hours) */
+        /*         } */
+        /*     } */
+        /*     Action { */ 
+        /*         text: qsTr("Until I turn it back on"); */
+        /*         icon.width: 0; */ 
+        /*         onTriggered: { */
+        /*             console.log(appSettings.mutedChannels) */
+        /*             appSettings.mutedChannels.push({ */
+        /*               name: "Foo" */
+        /*             }) */
+        /*             console.log(appSettings.mutedChannels) */
+        /*             //chatsModel.muteChannel(channelContextMenu.channelIndex, Constants.muteChatUntilUnmuted) */
+        /*         } */
+        /*     } */
+        /* } */
```

Comment:
As discussed offline, the context menu for mute durations is being left out for now